java9以上の環境で実行できない

```sh
build assemble
cd example/CloseToZero01
java -jar ../../build/libs/kGenProg-1.3.2.jar
```

上記コマンドを実行すると，以下のエラーがでる．
```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.RuntimeException: Cannot find extension class loader.
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:46)
	at io.reactivex.internal.operators.single.SingleCache.onError(SingleCache.java:146)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:60)
	at io.reactivex.internal.operators.single.SingleCache.onSuccess(SingleCache.java:134)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
	at io.reactivex.internal.operators.single.SingleJust.subscribeActual(SingleJust.java:30)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleCache.subscribeActual(SingleCache.java:65)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleCache.subscribeActual(SingleCache.java:65)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.Single.subscribe(Single.java:3481)
	at io.reactivex.Single.subscribe(Single.java:3402)
	at jp.kusumotolab.kgenprog.ga.variant.LazyVariant.subscribe(LazyVariant.java:27)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.createVariant(VariantStore.java:188)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.createInitialVariant(VariantStore.java:158)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.<init>(VariantStore.java:38)
	at jp.kusumotolab.kgenprog.KGenProgMain.run(KGenProgMain.java:66)
	at jp.kusumotolab.kgenprog.CUILauncher.launch(CUILauncher.java:65)
	at jp.kusumotolab.kgenprog.CUILauncher.main(CUILauncher.java:30)
Caused by: java.lang.RuntimeException: Cannot find extension class loader.
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:38)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:42)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:42)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.<init>(SkippingMemoryClassLoader.java:27)
	at jp.kusumotolab.kgenprog.project.test.TestThread.run(TestThread.java:98)
	at jp.kusumotolab.kgenprog.project.test.LocalTestExecutor.exec(LocalTestExecutor.java:29)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:57)
	... 21 more
Exception in thread "main" io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.RuntimeException: Cannot find extension class loader.
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.ConsumerSingleObserver.onError(ConsumerSingleObserver.java:46)
	at io.reactivex.internal.operators.single.SingleCache.onError(SingleCache.java:146)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onError(SingleMap.java:69)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:60)
	at io.reactivex.internal.operators.single.SingleCache.onSuccess(SingleCache.java:134)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
	at io.reactivex.internal.operators.single.SingleJust.subscribeActual(SingleJust.java:30)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleCache.subscribeActual(SingleCache.java:65)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleCache.subscribeActual(SingleCache.java:65)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.Single.subscribe(Single.java:3481)
	at io.reactivex.Single.subscribe(Single.java:3402)
	at jp.kusumotolab.kgenprog.ga.variant.LazyVariant.subscribe(LazyVariant.java:27)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.createVariant(VariantStore.java:188)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.createInitialVariant(VariantStore.java:158)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.<init>(VariantStore.java:38)
	at jp.kusumotolab.kgenprog.KGenProgMain.run(KGenProgMain.java:66)
	at jp.kusumotolab.kgenprog.CUILauncher.launch(CUILauncher.java:65)
	at jp.kusumotolab.kgenprog.CUILauncher.main(CUILauncher.java:30)
Caused by: java.lang.RuntimeException: Cannot find extension class loader.
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:38)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:42)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:42)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.<init>(SkippingMemoryClassLoader.java:27)
	at jp.kusumotolab.kgenprog.project.test.TestThread.run(TestThread.java:98)
	at jp.kusumotolab.kgenprog.project.test.LocalTestExecutor.exec(LocalTestExecutor.java:29)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:57)
	... 21 more
2019-04-01 16:33:37 [main] [INFO]  KGenProgMain - entered the era of 1st generation.
Exception in thread "main" java.lang.RuntimeException: Cannot find extension class loader.
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:38)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:42)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.findExtClassLoader(SkippingMemoryClassLoader.java:42)
	at jp.kusumotolab.kgenprog.project.test.SkippingMemoryClassLoader.<init>(SkippingMemoryClassLoader.java:27)
	at jp.kusumotolab.kgenprog.project.test.TestThread.run(TestThread.java:98)
	at jp.kusumotolab.kgenprog.project.test.LocalTestExecutor.exec(LocalTestExecutor.java:29)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:57)
	at io.reactivex.internal.operators.single.SingleCache.onSuccess(SingleCache.java:134)
	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
	at io.reactivex.internal.operators.single.SingleJust.subscribeActual(SingleJust.java:30)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleCache.subscribeActual(SingleCache.java:65)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleMap.subscribeActual(SingleMap.java:34)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.internal.operators.single.SingleCache.subscribeActual(SingleCache.java:65)
	at io.reactivex.Single.subscribe(Single.java:3495)
	at io.reactivex.Single.subscribe(Single.java:3481)
	at io.reactivex.Single.subscribe(Single.java:3402)
	at jp.kusumotolab.kgenprog.ga.variant.LazyVariant.subscribe(LazyVariant.java:27)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.createVariant(VariantStore.java:188)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.createInitialVariant(VariantStore.java:158)
	at jp.kusumotolab.kgenprog.ga.variant.VariantStore.<init>(VariantStore.java:38)
	at jp.kusumotolab.kgenprog.KGenProgMain.run(KGenProgMain.java:66)
	at jp.kusumotolab.kgenprog.CUILauncher.launch(CUILauncher.java:65)
	at jp.kusumotolab.kgenprog.CUILauncher.main(CUILauncher.java:30)
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -9,7 +9,7 @@ jobs:
   #   Execute `gradle test` to run unit tests on the target commit.
   run-unit-tests:
     docker:
-      - image: circleci/openjdk:8-jdk
+      - image: circleci/openjdk:11-jdk
     working_directory: ~/repo
     environment:
       JVM_OPTS: -Xmx3200m
@@ -58,7 +58,7 @@ jobs:
   #   Execute `gradle assemble` to generate a jar file on the target commit.
   build-artifacts:
     docker:
-      - image: circleci/openjdk:8-jdk
+      - image: circleci/openjdk:11-jdk
     steps:
       # ===== Initialize the environment
       - checkout

--- a/build.gradle
+++ b/build.gradle
@@ -17,8 +17,8 @@ apply plugin: 'idea'
 apply plugin: 'jacoco'
 
 // Set compiler version
-sourceCompatibility = '1.8'
-targetCompatibility = '1.8'
+sourceCompatibility = '1.11'
+targetCompatibility = '1.11'
 
 // Set default encoding
 compileJava.options.encoding = 'UTF-8'

--- a/src/main/java/jp/kusumotolab/kgenprog/project/test/SkippingMemoryClassLoader.java
+++ b/src/main/java/jp/kusumotolab/kgenprog/project/test/SkippingMemoryClassLoader.java
@@ -22,7 +22,7 @@ import java.net.URL;
  */
 public class SkippingMemoryClassLoader extends MemoryClassLoader {
 
-  final private ClassLoader extensionClassLoader;
+  final private ClassLoader delegationClassLoader;
 
   /**
    * コンストラクタ
@@ -31,24 +31,28 @@ public class SkippingMemoryClassLoader extends MemoryClassLoader {
    */
   public SkippingMemoryClassLoader(final URL[] urls) {
     super(urls);
-    extensionClassLoader = findExtClassLoader(getClass().getClassLoader());
+    delegationClassLoader = findDelegationClassLoader(getClass().getClassLoader());
   }
 
   /**
-   * クラスローダの親子関係を探索して，委譲先となるextensionClassLoaderを探す．
+   * クラスローダの親子関係を探索して，委譲先となるExtension/PlatformClassLoaderを探す．
    * 
    * @param cl
    * @return
    */
-  private ClassLoader findExtClassLoader(final ClassLoader cl) {
+  private ClassLoader findDelegationClassLoader(final ClassLoader cl) {
     if (null == cl) {
       throw new RuntimeException("Cannot find extension class loader.");
     }
-    if (!cl.toString()
-        .contains("$ExtClassLoader@")) { // clが取り出したいExtClassLoaderではない場合，
-      return findExtClassLoader(cl.getParent()); // 再帰的に親を探す
+    // (#600) patch for greater than jdk9
+    // 対象の名前が ExtensionClassLoader (jdk8) or PlatformClassLoader(>jdk9) ならOK．
+    final String name = cl.toString();
+    if (name.contains("$ExtClassLoader@") || //
+        name.contains("$PlatformClassLoader@")) {
+      return cl;
     }
-    return cl;
+    // さもなくば再帰的に親を探す
+    return findDelegationClassLoader(cl.getParent());
   }
 
   /**
@@ -73,7 +77,7 @@ public class SkippingMemoryClassLoader extends MemoryClassLoader {
       if (null == c) {
         try {
           // Second, try to load using extension class loader
-          c = extensionClassLoader.loadClass(name);
+          c = delegationClassLoader.loadClass(name);
 
           // Don't delegate to the parent.
           // c = parent.loadClass(name);

